### ISSUE TITLE
Not enough permissions?

### ISSUE THREAD

**Issue by adamgotterer** [ericentin/exexec#8]
Inside a docker container I'm using `exexec` to execute a command. That command runs fine as that user directly in the terminal. But when I try and run the exact same command with `exexec` it says `"Pid 64: cannot execute '(null)': Bad address"` which from some research seems to be some sort of permission issue. Any ideas?

**Comment by adamgotterer** [comment]
I got it working.

Simple fix for docker: `ENV SHELL /bin/sh`

Turns out it was executing in the wrong shell and the exec command couldn't find the program.

**Comment by j-manu** [comment]
Thank you for sharing the solution! I struggled with this for hours. Do you know why this happens despite giving the full path to the executable?

**Comment by adamgotterer** [comment]
The clue that fixed this for me was: https://github.com/saleyn/erlexec/blob/9eb236bc7526e58db9e9ec4109c134bfd65105d8/src/overview.edoc#L104

I'll be honest, I'm not entirely sure why it doesn't work. I've spent a lot of time trying to understand this one and I just can't seem to figure it out. These are my theories:

1) It was never clear to me what wasn't found. Was it the file? The shell? So is it's possible it is looking for a shell that doesn't exist?
2) In my case the default user shell is /bin/sh. Maybe erlexec is trying to use a shell that has the PATH setup incorrectly.
3) Permissions for execution from SH and Bash differ. Maybe they are wrong (although I've played with the permissions extensively).

If someone else has a better understanding I'm definitely curious...
